manage_breast_screening/auth/services.py

```diff
@@ -0,0 +1,71 @@
+import logging
+from typing import Any, Dict
+
+from authlib.jose import JoseError, jwt
+from authlib.jose.errors import (
+    ExpiredTokenError,
+    InvalidClaimError,
+    InvalidTokenError,
+    MissingClaimError,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class InvalidLogoutToken(Exception):
+    """Raised when a CIS2 back-channel logout token is invalid."""
+
+    def __init__(self, cause=None):
```

Comment:
I think this overriding is probably unnecessary here - when you do `raise ... from` the original exception is already stored, and the stack trace will contain the original exception at the end:

```
Exception: abc

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
Exception: def
```